activemodel/test/cases/secure_password_test.rb

```diff
@@ -31,6 +33,11 @@ class SecurePasswordTest < ActiveModel::TestCase
     assert_not_respond_to @visitor, :valid?
   end
 
+  test "create a new user and bypassing the validations" do
+    @person_skipping_secure_password_validations.password = ""
+    assert @person_skipping_secure_password_validations.valid?
+  end
```

Comment:
With some Ruby magic, we can test this without an extra (statically defined) model:

```suggestion
  test "supports conditional validation" do
    model_class = Struct.new(:requires_password, :password_digest) do
      include ActiveModel::SecurePassword
      has_secure_password validations: { if: :requires_password }
    end

    assert_predicate model_class.new(false), :valid?
    assert_predicate model_class.new(true), :invalid?
  end
```

Reply:
🙏 Thanks, yes indeed a Struct will work just fine here.